devops/scripts/build_zstd_1_5_6_ub24.sh

```diff
@@ -0,0 +1,105 @@
+#!/bin/bash
+
+# Function to check if the OS is Ubuntu 24
+check_os() {
+    . /etc/os-release
+    if [[ "$NAME" != "Ubuntu" || "$VERSION_ID" != "24.04" ]]; then
+        echo "Warning: This script has only been tested with Ubuntu 24."
+    fi
+}
+
+# Function to install packages with or without sudo
+install_packages() {
+    if [ "$USE_SUDO" = true ]; then
+        sudo apt-get update
+        sudo apt-get install -y build-essential wget
+    else
+        apt-get update
+        apt-get install -y build-essential wget
+    fi
+}
+
+# Function to uninstall libzstd-dev if installed
+uninstall_libzstd_dev() {
+    if dpkg -l | grep -q libzstd-dev; then
+        if [ "$USE_SUDO" = true ]; then
+            sudo apt-get remove -y libzstd-dev
+        else
+            apt-get remove -y libzstd-dev
+        fi
+    fi
+}
+
+# Function to build and run a simple test program
+build_test_program() {
+    cat <<EOF > test_zstd.c
+      #include <zstd.h>
+      int main() {
+        ZSTD_CCtx* cctx = ZSTD_createCCtx();
+        ZSTD_freeCCtx(cctx);
+        return 0;
+      }
+EOF
+
+    # Try to use zstd's static library with -fPIC
+    gcc test_zstd.c -lzstd -fPIC -shared
+    if [ $? -ne 0 ]; then
+        echo "zstd installation verification failed."
+    else
+        echo "zstd installation verification passed."
+    fi
+
+    # There won't be a.out file if verification failed.
+    rm test_zstd.c a.out || true
+}
+
+# Check the OS
+check_os
+
+# Set USE_SUDO to true or false based on your preference
+USE_SUDO=true
```

Comment:
if we're always using sudo can we just use it directly without this var?

Reply:
I have a slight preference towards keeping this variable because it makes it a lot easier for me to test this script internally using podman (which requires not using `sudo`). I can remove this variable if you strongly feel so.

Reply:
ah no if it has a use thats totally fine, thx